spring-core/src/main/java/org/springframework/util/ClassUtils.java

```diff
@@ -658,30 +658,35 @@ else if (candidates.isEmpty()) {
 	 * @see Class#getMethod
 	 */
 	@Nullable
-	public static Method getMethodIfAvailable(Class<?> clazz, String methodName, @Nullable Class<?>... paramTypes) {
+	public static Method getMethodIfAvailable(Class<?> clazz, String methodName,
+			@Nullable Class<?>... paramTypes) {
 		Assert.notNull(clazz, "Class must not be null");
 		Assert.notNull(methodName, "Method name must not be null");
-		if (paramTypes != null) {
-			try {
-				return clazz.getMethod(methodName, paramTypes);
-			}
-			catch (NoSuchMethodException ex) {
-				return null;
-			}
+		if (paramTypes == null) {
+			return findSingleMethod(clazz, methodName);
 		}
-		else {
-			Set<Method> candidates = new HashSet<>(1);
-			Method[] methods = clazz.getMethods();
-			for (Method method : methods) {
-				if (methodName.equals(method.getName())) {
-					candidates.add(method);
-				}
+		Method[] methods = clazz.getMethods();
+		for (Method method : methods) {
+			if (method.getName().equals(methodName)
+					&& Arrays.equals(method.getParameterTypes(), paramTypes)) {
```

Comment:
May I suggest a pre-check for `getParameterCount() == paramTypes.length` in order to avoid possible allocations coming from `getParameterTypes()`?

Reply:
Good call. Will rework the PR. Thanks!